src/librustdoc/clean/auto_trait.rs

```diff
@@ -1435,9 +1435,7 @@ impl<'a, 'tcx, 'rcx> AutoTraitFinder<'a, 'tcx, 'rcx> {
     // involved (impls rarely have more than a few bounds) means that it
     // shouldn't matter in practice.
```

Comment:
Also probably not this PR, but this comment makes me wonder if it would be worth adding `if len < 2 { return }` to the checks excerpted below, since sort_by_cached_key is currently a pessimization in the len==1 case, as it'll compute the key and allocate, which ordinary sort_by_key [wouldn't](https://github.com/rust-lang/rust/blob/1c5283b47210e51978a32c6e76d0fb71b03e11f4/src/liballoc/slice.rs#L2228-L2236).

https://github.com/rust-lang/rust/blob/1c5283b47210e51978a32c6e76d0fb71b03e11f4/src/liballoc/slice.rs#L1402-L1406

Reply:
That's another good point: I'll add that extra check in! (`len == 2` could also be special cased to not allocate, as only one comparison will be executed, but that's a less likely case to encounter.)